libs/langchain-core/src/tracers/tracer_langchain.ts

```diff
@@ -43,6 +47,32 @@ export interface LangChainTracerFields extends BaseCallbackHandlerInput {
   replicas?: RunTreeConfig["replicas"];
 }
 
+/**
+ * Extract usage_metadata from generations.
+ *
+ * Iterates through generations to find and return the first usage_metadata
+ * found in a message. This is typically present in chat model outputs.
+ */
+function _getUsageMetadataFromGenerations(
+  generations: Generation[][]
+): UsageMetadata | undefined {
+  for (const generationBatch of generations) {
+    for (const generation of generationBatch) {
+      if ("message" in generation) {
+        const { message } = generation as { message: unknown };
+        if (
+          isBaseMessage(message) &&
+          isAIMessage(message) &&
+          message.usage_metadata !== undefined
+        ) {
+          return message.usage_metadata;
```

Comment:
Should we only do this in the same scenario as when we flatten?

Not that anyone should have multiple generations anymore but would be unexpected if we just took the first value

Proper thing would actually be to sum them I think?